src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -234,7 +234,9 @@ def load_arguments(self, _):
         c.argument('identifier_uri', help='graph application identifier, must be in uri format')
         c.argument('spn', help='service principal name')
         c.argument('upn', help='user principal name, e.g. john.doe@contoso.com')
-        c.argument('query_filter', options_list=['--filter'], help='OData filter, e.g. --filter "displayname eq \'test\' and servicePrincipalType eq \'Application\'"')
+        c.argument('query_filter', options_list=['--filter'],
+                   help='The $filter OData query parameter to retrieve a subset of a collection. '
+                        'For more information, see https://learn.microsoft.com/graph/filter-query-parameter')
```

Comment:
Another thing is **how we can better convey that quoting rules should be considered in different shells**. Especially, when there are spaces in the value, the value should be quoted.

Currently `--query` doesn't mention quoting rules.

```
    --query            : JMESPath query string. See http://jmespath.org/ for more information and
                         examples.
```